### DISCUSSION TITLE
diesel 2.0 schema.rs file build not end even after 20 mins.

### DISCUSSION THREAD

**Discussion by forest1102** [diesel-rs/diesel#3410] [Q&A] [answered]
Hello, suppose I have 423 tables, and each table has id column and 61 other columns. 
[sample huge table schema code](https://gist.github.com/forest1102/be2b1acbd2c0148bd6c1775888c865bf)
While with diesel 1.4, the building time of this code is ~4m26s, with diesel 2.0, the build never ends even after 20mins. 
Can you help me why this build never ends? 

## Environment

- docker image `rust:latest`
  ```
  $ rustc --version
    rustc 1.65.0 (897e37553 2022-11-02)
  $ cat /proc/version
    Linux version 5.10.16.3-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Apr 2 22:23:49 UTC 2021
  ```

- .cargo/config
  
  ```toml
  [target.x86_64-pc-windows-gnu]
  linker = "x86_64-w64-mingw32-gcc"
  [target.x86_64-unknown-linux-gnu]
  rustflags = [
      "-C", "link-arg=-fuse-ld=/usr/local/bin/mold",
  ]
  linker = "clang"
  ```

**Comment by Ten0** [comment, marked as answer]
We have updated the trait system to improve the extent to which Diesel can check the validity of queries. It used to allow some invalid query patterns, and these are now largely prohibited. (notably with regards to joins, nullability knowledge...)
Unfortunately, that did come at the cost of increasing compilation time.
There has already been some work on making compilation time lower (#3223 notably).

It seems your issue would require investigation similar to what was done in 3223, essentially following [this guide](https://blog.rust-lang.org/inside-rust/2020/02/25/intro-rustc-self-profile.html), to get a sense of where the time is spent.

**Reply by weiznich** [reply]
I personally do currently not have the capacity to sit down to debug this issue. If someone provides the corresponding profiles I'm happy to answer questions. Otherwise I also suggest filling a bug report in the rustc repository. There is likely room there to make more efficiently cache the relevant trait bounds.

**Reply by weiznich** [reply]
I cannot reproduce the described behaviour with the provided example. The provided code builds in ~8minutes for me on my rather slow laptop.

**Reply by forest1102** [reply]
@weiznich 
Hello, I updated the environment where this problem occurs. 
I used docker with the image `rust:latest`.
Can you tell me the execution environment?

**Reply by weiznich** [reply]
I tested this with a local rustc on ubuntu with the following rustc versions:

* 1.65.0
* 1.64.0

**Reply by forest1102** [reply]
Sorry, then I have no idea why this happens. 
Can you tell me possible reasons why this happens if you have?

**Reply by weiznich** [reply]
I cannot help you here anymore without having a reproducible example.

**Reply by forest1102** [reply]
Sorry for bothering you, I somehow fixed it.
